### ISSUE TITLE
why the mAP decreases when i train the model by the following step?

### ISSUE THREAD

**Issue by racecdj** [naoto0804/cross-domain-detection#7]
I'm trying to reconstruct the experiment. The step i do is as the  following：
1.DT
`python train_model.py --root datasets/dt_clipart/VOC2007 --root datasets/dt_clipart/VOC2012 --subset train --result result/dt_clipart --det_type ssd300 --data_type clipart --gpu 0 --max_iter 500 --load models/clipart_dt_ssd300 --eval_root datasets/clipart`
2.PL
（1）get labels
`python pseudo_label.py --root datasets/clipart --data_type clipart --det_type ssd300 --gpu 0 --load result/dt_clipart/model_iter_10000 --result datasets/dt_pl_clipart`
（2）fine-tune
`python train_model.py --root datasets/dt_pl_clipart --subset train --result result/dt_pl_clipart --det_type ssd300 --data_type clipart --gpu 0 --load result/dt_clipart/model_iter_10000 --eval_root datasets/clipart`

Questions:
1.In step1, there is no test.txt in the path:datasets/dt_datasets/dt_clipart/VOC2012/ImageSets/Main.
when i do step1, the code throwed an error, then i looked the content of the files in folder Main. Finally, i choose val.txt as the test.txt to evaluate the mAP in step1.
Is this way correct?

2.From the result of doing the three steps, i find the mAP in step1 is 0.6834. And after step3, the final mAP is 0.435. The iteration is 10000, batchsize is 16. I want to know if I made a mistake in some steps ?

3.How can i use the extra dataset in the experiment of watercolor part ?

I look forward to hearing from you.

**Comment by racecdj** [comment]
The result of the step1:
mAP:
![map](https://user-images.githubusercontent.com/33973768/50422656-e872fa80-0886-11e9-98c4-55a4c551249b.png)
Loss:
![loss](https://user-images.githubusercontent.com/33973768/50422658-ed37ae80-0886-11e9-835c-bff31ef71465.png)
The result of the step3:
mAP:
![map](https://user-images.githubusercontent.com/33973768/50422660-f294f900-0886-11e9-97ff-b19a7cd69122.png)
Loss
![loss](https://user-images.githubusercontent.com/33973768/50422661-f58fe980-0886-11e9-8c85-1509d8fa02c8.png)

**Comment by racecdj** [comment]
Can someone help me solve this problem?

**Comment by naoto0804** [comment]
Sorry I'm quite busy with submission, please wait for a moment..

**Comment by naoto0804** [comment]
Thank you for letting me know. I've updated the README.
A1. Please try again using `--subset trainval` and `--eval_root datasets/clipart` in step1 (DT). 
A2. I'm afraid you evaluate your model on the dataset that you train on (68 in AP is too high). Please retry following A1. Due to the small dataset size, the performance might be unstable, I'm sorry.
A3. IDs of the extra images are in `ImageSets/Main/extra.txt` for each dataset. In the training phase, please use `--subset trainextra` instead of `--subset trainextra` to add extra images.

**Comment by racecdj** [comment]
hello，I'm doing the latest steps. During the step3, I met a new problem. `Memory Error`
I haven't seen it in previous experiments.
Here is the screenshoot.
<img width="904" alt="1" src="https://user-images.githubusercontent.com/33973768/50723680-d95c3a00-111b-11e9-81e2-fa9c9af5c8ba.png">

**Comment by naoto0804** [comment]
I have not faced such problem, is it reproducible?

**Comment by racecdj** [comment]
I met another question:
with the iter time increases, the map of clipart is like this. I just follow the command as you posted in ReadMe.md
<img width="525" alt="wx20190117-231051 2x" src="https://user-images.githubusercontent.com/33973768/51327870-30afc200-1aad-11e9-8b3b-55e45a053898.png">

And the result of DL+PL in  watercolor part is not less than ideal. You can see the map picture. Is this caused by the small dataset?
<img width="525" alt="wx20190117-231308 2x" src="https://user-images.githubusercontent.com/33973768/51328045-79677b00-1aad-11e9-8184-21bdb49babc4.png">

**Comment by naoto0804** [comment]
Unfortunately, I have no idea other than that. Constructing a bigger dataset is a promising way.

**Comment by racecdj** [comment]
Thanks for your answer. Last question, how can I change det_type from ssd300 to yolov2 or faster r-cnn?

**Comment by naoto0804** [comment]
We push support for FasterRCNN (based on VGG16) in the master branch. You can specify the detector by `--det_type` option. You can choose `ssd300`, `ssd512`, and `faster` (Faster RCNN on VGG backbone). If you need the trained models in the paper, please let me know again.

For YOLO, we did our experiment using the author's original code and it's hard to share everything because of complexity in pre-processing and post-processing, I'm sorry.
